lib/SILOptimizer/Transforms/SimplifyCFG.cpp

```diff
@@ -916,16 +916,44 @@ static bool couldRemoveRelease(SILBasicBlock *SrcBB, SILValue SrcV,
   return IsReleaseOfDest;
 }
 
+/// Returns true if any instruction in \p block may write memory.
+static bool blockMayWriteMemory(SILBasicBlock *block) {
+  return std::any_of(block->begin(), block->end(), [](SILInstruction &inst) {
+    return inst.mayWriteToMemory();
+  });
+}
+
+// Returns true if \p block contains an injected an enum case into \p enumAddr
+// which is valid at the end of the block.
+static bool hasInjectedEnumAtEndOfBlock(SILBasicBlock *block, SILValue enumAddr) {
+  for (SILInstruction &inst : llvm::reverse(*block)) {
+    if (auto *injectInst = dyn_cast<InjectEnumAddrInst>(&inst)) {
```

Comment:
Same potential issue here of quadratic compile time with no potential benefit. A simple cutoff would suffice.